src/material/core/common-behaviors/error-state.ts

```diff
@@ -62,10 +65,7 @@ export class _ErrorStateTracker {
     const parent = this._parentFormGroup || this._parentForm;
     const matcher = this.matcher || this._defaultMatcher;
     const control = this.ngControl ? (this.ngControl.control as AbstractControl) : null;
-    // Note: the null check here shouldn't be necessary, but there's an internal
-    // test that appears to pass an object whose `isErrorState` isn't a function.
-    const newState =
-      typeof matcher?.isErrorState === 'function' ? matcher.isErrorState(control, parent) : false;
+    const newState = matcher?.isErrorState(control, parent);
```

Comment:
This will be `boolean | undefined` while previously it was `boolean`.

Reply:
Good catch. I fixed this so that newState will always be a boolean and not undefined. I'm surprised the compiler didn't catch this mistake.